packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.test.js

```diff
@@ -1,14 +1,572 @@
+				const calendar = {
+					year: 2018,
+					monthIndex: 2,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(getFirstRowCellText(wrapper, 4)).toBe('1');
+			});
+
+			it('should have sunday in the 7th column for april 2018', () => {
+				const calendar = {
+					year: 2018,
+					monthIndex: 3,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(getFirstRowCellText(wrapper, 7)).toBe('1');
+			});
+		});
+
+
+		describe('last date in the correct day column (Monday as first column)', () => {
+			function getLastRowCellText(wrapper, column) {
+				return wrapper
+					.find('.theme-calendar-body-row')
+					.last()
+					.find('.theme-calendar-item')
+					.at(column - 1)
+					.find('DayPickerAction')
+					.prop('label');
+			}
+
+			it('should have monday in the 1st column for april 2018', () => {
+				const calendar = {
+					year: 2018,
+					monthIndex: 3,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(getLastRowCellText(wrapper, 1)).toBe('30');
+			});
+
+			it('should have thursday in the 4th column for may 2018', () => {
+				const calendar = {
+					year: 2018,
+					monthIndex: 4,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(getLastRowCellText(wrapper, 4)).toBe('31');
+			});
+
+			it('should have sunday in the 7th column for september 2018', () => {
+				const calendar = {
+					year: 2018,
+					monthIndex: 8,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(getLastRowCellText(wrapper, 7)).toBe('30');
+			});
+		});
+
+		describe('number of weeks displayed', () => {
+			function nbOfWeeksRendered(wrapper) {
+				return wrapper.find('.theme-calendar-body-row').length;
+			}
+
+			it('should have 4 weeks for february 2010', () => {
+				const calendar = {
+					year: 2010,
+					monthIndex: 1,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(nbOfWeeksRendered(wrapper)).toBe(4);
+			});
+
+			it('should have 5 weeks for june 2018', () => {
+				const calendar = {
+					year: 2018,
+					monthIndex: 5,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(nbOfWeeksRendered(wrapper)).toBe(5);
+			});
+
+			it('should have 6 weeks for october 2017', () => {
+				const calendar = {
+					year: 2017,
+					monthIndex: 9,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(nbOfWeeksRendered(wrapper)).toBe(6);
+			});
+		});
+
+		describe('contiguous ordered numbers', () => {
```

Comment:
This section is implicitly tested in the snapshot of the 1st test. You can remove that (and don't need to test that for multiple days)

Reply:
✔

Reply:
The only differences between your cases are the number of day.
You can remove the `continuous` part in description and assertion. This is tested in the snapshot. Here you change the month/year to change the number of days, just assert that.

Reply:
✔